apps/04TowersOfHanoi.js

```diff
@@ -16,24 +16,64 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack, endStack) {
     // Your code here
+    // pop
+    var piece = stacks[startStack].pop();
+    // push
+    stacks[endStack].push(piece);
 
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
     // Your code here
+    var startStackArray = stacks[startStack];
+    var endStackArray = stacks[endStack];
+      //var lastElementOfStart = startStackArray[startStackArray.lenghth - 1];
+      //  var lastElementofEnd = endStack[endStack.lenghth - 1];
+    if (stacks[startStack].length === 0){
+      // Start stack is empty - Empty stack invalid
+      return false;
+    }
+
+    if (startStackArray.length === 0){
+          return false;
+    } else if (endStackArray.length === 0){
+          return true;
+    } else {
+          return startStackArray[startStackArray.length - 1] < endStackArray[endStackArray.length - 1];
+    }
+//    if (stacks[endStack].lenghth === 0){
+//      return true;
+//    }
+
+    // if(stacks[startStack][stack[startStack].length] < stacks[endStack][stacks][endStack].length - 1){
+    //     return true;
+    // }else{
+    //   return false;
+    // }
 
 }
 
 function checkForWin() {
     // Your code here
-
+      if (stacks.b.length >= 4 || stacks.c.length >= 4){
+            console.log("YOU WON! WAY TO GO!!");
+          return true;
+      } else {
+          return false;
+      }
 }
 
 function towersOfHanoi(startStack, endStack) {
     // Your code here
-
+    if (isLegal(startStack, endStack) === true){
```

Comment:
checking === true is a bit redundant. Remember === comparisions result in a boolean value. But isLegal() ALREADY returns a boolean. So just use it! No need to compare it to another boolean.